src/plugin.ts

```diff
@@ -20,89 +13,70 @@ const getAdapter = (): AdapterInterface => {
+      }
+
+      const adapter = getAdapter()
+      await adapter.upload(data.filename, uploadedFile)
     }
 
-    const adapter = getAdapter()
-    await adapter.upload(data.filename, uploadedFile)
+    return data
   }
-
-  return data
-}
-
-export const deleteHook: CollectionAfterDeleteHook = async ({ doc }) => {
-  const adapter = getAdapter()
-  await adapter.delete(doc.filename)
 }
 
-export function withCloudStorage (
-  collection: SanitizedCollectionConfig,
-  adapter: AdapterInterface,
-  getUrl: (filename: string) => string,
-): SanitizedCollectionConfig {
-  adapterInstance = adapter
-
-  collection.fields = [
-    ...collection.fields,
-    {
-      label: 'Cloud Storage URL',
-      name: 'cloudStorageUrl',
-      type: 'text',
-      admin: {
-        readOnly: true,
-      },
-      hooks: {
-        beforeChange: [
-          (): undefined => undefined,
-        ],
-        afterRead: [
-          ({ data }): string => {
-            return getUrl(String(data.filename))
-          },
-        ],
-      },
-    },
-  ]
-
-  const {
-    beforeChange = [],
-    afterDelete = [],
-  } = collection.hooks || {}
-
-  collection.hooks = {
-    ...collection.hooks,
-    beforeChange: [
-      ...beforeChange,
-      uploadHook,
-    ],
-    afterDelete: [
-      ...afterDelete,
-      deleteHook,
-    ],
+export const deleteHook: CollectionAfterDeleteHook = async (args) => {
+  if (args) {
+    const { doc } = args
+    const adapter = getAdapter()
+    await adapter.delete(doc.filename)
   }
-
-  collection.upload.adminThumbnail = ({ doc }: { doc: { cloudStorageUrl: string } }) => doc.cloudStorageUrl
-
-  return collection
 }
 
-
 const cloudStorage = (
   adapter: AdapterInterface,
-  getUrl: (filename: string) => string,
+  uploadCollectionModifiers?: S3PluginCollectionModifiers
 ) => {
+  adapterInstance = adapter
   return (incommingConfig: SanitizedConfig): SanitizedConfig => {
     const config: SanitizedConfig = {
       ...incommingConfig,
       collections: incommingConfig.collections.map(collection => {
         if (typeof collection.upload === 'object') {
-          return withCloudStorage(collection, adapter, getUrl)
+          if (Array.isArray(uploadCollectionModifiers?.fields) && uploadCollectionModifiers?.fields.length) {
+            collection.fields = [
+              ...collection.fields,
+              ...uploadCollectionModifiers.fields
+            ]
+          }
+
+          const {
+            beforeChange = [],
+            afterDelete = [],
+          } = collection.hooks || {}
+
+          collection.hooks = {
+            ...collection.hooks,
+            beforeChange: [
+              ...beforeChange,
+              uploadHook,
+            ],
+            afterDelete: [
+              ...afterDelete,
+              deleteHook,
+            ],
+          }
+
+          if (uploadCollectionModifiers?.adminThumbnail && typeof collection?.upload?.adminThumbnail === 'undefined') {
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/richardvanbergen/payload-plugin-cloud-storage/pullRequest?prid=7829468)
```suggestion
          if (uploadCollectionModifiers?.adminThumbnail && typeof collection?.upload?.adminThumbnail === "undefined") {
```